packages/@aws-cdk/aws-efs/lib/efs-file-system.ts

```diff
@@ -272,11 +272,13 @@ export class FileSystem extends Resource implements IFileSystem {
       defaultPort: ec2.Port.tcp(FileSystem.DEFAULT_PORT),
     });
 
+    this.mountTargetsAvailable = [];
+    if (props.vpcSubnets === undefined) return;
```

Comment:
let's add a test for this. should there be a warning or error to indicate that mounts targets are not created